packages/esm-service-queues-app/src/patient-queue-header/patient-queue-header.component.tsx

```diff
@@ -15,16 +15,24 @@ import styles from './patient-queue-header.scss';
 
 const PatientQueueHeader: React.FC<{ title?: string }> = ({ title }) => {
   const { t } = useTranslation();
-  const { queueLocations } = useQueueLocations();
+  const { queueLocations, isLoading, error } = useQueueLocations();
   const userSession = useSession();
   const userLocation = userSession?.sessionLocation?.display;
   const currentQueueLocationName = useSelectedQueueLocationName();
 
-  const handleQueueLocationChange = ({ selectedItem }) => {
+  const handleQueueLocationChange = useCallback((selectedItem) => {
     updateSelectedQueueLocationUuid(selectedItem.id);
     updateSelectedQueueLocationName(selectedItem.name);
     updateSelectedServiceName('All');
-  };
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading...</p>; // Show loading state while fetching data
+  }
```

Comment:
I agree with @vasharma05 about this. You can add the check for loading state only within the dropdown and add the dropdown skeleton from carbon 

https://react.carbondesignsystem.com/?path=/story/components-dropdown--skeleton

Reply:
Done 
<img width="1438" alt="Screenshot 2024-01-11 at 12 33 22 PM" src="https://github.com/openmrs/openmrs-esm-patient-management/assets/94985341/4df2cf16-dd8a-4187-8428-800e06051b6a">